File: puzzle11_change_coins/coin_change.rb
class CoinChange
  def initialize(coin_types)
    @coin_types = coin_types
  end

  def make(amount)
    @coin_types.sort.reverse.inject([]) do |result, coin_type|
      while amount >= coin_type
        result << coin_type
        amount -= coin_type
      end
      result
    end
  end
end

File: puzzle11_change_coins/CoinChange.js
function CoinChange(coinTypes) {
  this.coinTypes = coinTypes;
  this.make = function(amount) {
    var result = [];
    var coinTypes = this.coinTypes.sort(function(a,b) { return b-a; });
    $(coinTypes).each(function() {
      while (amount >= this) {
        result.push(this);
        amount -= this;
      }
    });
    return result;
  };
}
